src/test/java/io/r2dbc/postgresql/codec/IntervalUnitTests.java

```diff
@@ -0,0 +1,536 @@
+                new Object[]{"P8Y2M3DT4H5M6.545S", Interval.of(8, 2, 3, 4, 5, 6, 545000)},
+                new Object[]{"P8Y2M3DT4H5M6.54567S", Interval.of(8, 2, 3, 4, 5, 6, 545670)},
+                new Object[]{"PT0S", Interval.ZERO},
+                new Object[]{"P-1Y-2M3DT-4H-5M-6S", Interval.of(-1, -2, 3, -4, -5, -6, 0)}
+        );
+    }
+
+    static Stream<Object[]> intervalValues() {
+        return Stream.of(
+                new Object[]{Interval.ZERO, "0 yr 0 mon 0 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.ofYears(178)), "178 yr 0 mon 0 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.ofMonths(11)), "0 yr 11 mon 0 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.ofWeeks(4)), "0 yr 0 mon 28 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.ofDays(25)), "0 yr 0 mon 25 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.of(58, 4, 0)), "58 yr 4 mon 0 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.of(3413, 0, 9)), "3413 yr 0 mon 9 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.of(0, 2, 3)), "0 yr 2 mon 3 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.of(99, 9, 9)), "99 yr 9 mon 9 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.ofYears(-941592)), "-941592 yr 0 mon 0 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.ofMonths(-10)), "0 yr -10 mon 0 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.ofDays(-17)), "0 yr 0 mon -17 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.of(-523, 5, -21)), "-523 yr 5 mon -21 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Period.of(-89, 0, 5)), "-89 yr 0 mon 5 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Duration.ofDays(-98)), "0 yr 0 mon -98 day 0 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Duration.ofHours(6)), "0 yr 0 mon 0 day 6 hr 0 min 0.0 sec"},
+                new Object[]{Interval.of(Duration.ofMinutes(35)), "0 yr 0 mon 0 day 0 hr 35 min 0.0 sec"},
+                new Object[]{Interval.of(Duration.ofSeconds(15)), "0 yr 0 mon 0 day 0 hr 0 min 15.0 sec"},
+                new Object[]{Interval.of(Duration.ofMillis(753)), "0 yr 0 mon 0 day 0 hr 0 min 0.753 sec"},
+                new Object[]{Interval.of(Duration.ofNanos(1000)), "0 yr 0 mon 0 day 0 hr 0 min 0.000001 sec"},
+                new Object[]{Interval.of(Duration.ofSeconds(987654321)), "0 yr 0 mon 11431 day 4 hr 25 min 21.0 sec"},
+                new Object[]{Interval.of(Duration.ofSeconds(443121)), "0 yr 0 mon 5 day 3 hr 5 min 21.0 sec"},
+                new Object[]{Interval.of(Period.of(45, 6, 9), Duration.ofSeconds(443121)), "45 yr 6 mon 14 day 3 hr 5 min 21.0 sec"},
```

Comment:
There seems something to be off in the parser. The resulting `Interval` reports 45 hours.